src/controllers/authController.js

```diff
@@ -137,21 +142,15 @@ const resetPassword = async (req, res) => {
       return res.status(404).json({ error: `Token expired` })
     }
 
-    bcrypt.hash(req.body.password, 8, async (gotError, hashedPass) => {
-      if (gotError) {
-        return res.status(400).send({
-          error: 'register failed'
-        })
-      }
-
-      user.password = hashedPass
-
+    try {
+      user.password = await bcrypt.hash(req.body.password, 8)
```

Comment:
```suggestion
      user.password = await bcrypt.hash(req.body.password, 8)
```
Aqui o user.password não existe, porque você não deu o select no método findOne acima

Reply:
É necessário fazer o select do password só se for feito um get no caso de estar setando não precisa.